### PR TITLE
Fix BA plugin after latest game update.

### PR DESCRIPTION
**Pull request by Jacoblairm**
No longer rely on the previously used "icons" to determine the role, instead use the role widget currently loaded. Tested and working https://i.imgur.com/x0JnDsD.png

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/barbarianassault/BarbarianAssaultPlugin.java [hunk 1]
```diff
@@ -115,13 +110,38 @@ protected void shutDown() throws Exception
 	@Subscribe
 	public void onWidgetLoaded(WidgetLoaded event)
 	{
-		if (event.getGroupId() == WidgetID.BA_REWARD_GROUP_ID)
```
Comment (Hydrox6) [inline runelite-client/src/main/java/net/runelite/client/plugins/barbarianassault/BarbarianAssaultPlugin.java]: Why have you removed the stuff that handles the Wave Times option?

Comment (Jacoblairm) [reply runelite-client/src/main/java/net/runelite/client/plugins/barbarianassault/BarbarianAssaultPlugin.java]: I haven't, I just moved it into the switch statement.

Comment (Hydrox6) [reply runelite-client/src/main/java/net/runelite/client/plugins/barbarianassault/BarbarianAssaultPlugin.java]: Ah, that's a confusing diff. Nvm then

**Comment by Adam-** [comment]
Is the `gameTime = null` fix from #8622 important? should it be here?

**Comment by Jacoblairm** [comment]
> Is the `gameTime = null` fix from #8622 important? should it be here?

Not too important but I just committed the fix.

**Comment by Twiglet1022** [comment]
I've just tried it this out at some BA and it works great except for one issue which is that the timer resets to 30 secs when you close the horn of glory interface. It always disappeared when the horn of glory was up, but it would remember the time for when you closed it again.

It's pretty minor tbh because I doubt many people use the horn, but I find it useful sometimes.

I think it might be because you're now relying on the widget, so the way #8622 does it might be better because you still have the horn in your inventory when you open that interface.

**Comment by Adam-** [comment]
@Jacoblairm what does it fix?

**Comment by Jacoblairm** [comment]
> I've just tried it this out at some BA and it works great except for one issue which is that the timer resets to 30 secs when you close the horn of glory interface. It always disappeared when the horn of glory was up, but it would remember the time for when you closed it again.
> 
> It's pretty minor tbh because I doubt many people use the horn, but I find it useful sometimes.
> 
> I think it might be because you're now relying on the widget, so the way #8622 does it might be better because you still have the horn in your inventory when you open that interface.

Fixed this issue by checking if the currentRound has already been set during the round.

**Comment by Adam-** [comment]
I meant, what is `gameTime = null` fixing? What happens if you don't add it?

**Comment by Jacoblairm** [comment]
@Adam- https://i.imgur.com/fNbitFp.png occurs when you complete a tutorial and waveTime is not null (ie - after completing at least one wave). I just realized `gameTime = null` doesn't properly fix it; I have remediated the bug in the latest commit.

**Comment by Adam-** [comment]
Is there a reason why this is waiting until you enter the following wave to tell you your previous waves time?

File: runelite-client/src/main/java/net/runelite/client/plugins/barbarianassault/BarbarianAssaultPlugin.java [hunk 2]
```diff
@@ -195,7 +177,7 @@ public void onVarbitChanged(VarbitChanged event)
 			{
 				overlay.setCurrentRound(null);
 
-				if (config.waveTimes() && gameTime != null)
+				if (config.waveTimes() && gameTime != null && client.isInInstancedRegion())
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/barbarianassault/BarbarianAssaultPlugin.java]: Is this to try and avoid messages from the tutorials? How does that work as the tutorials are instanced?

Comment (Jacoblairm) [reply runelite-client/src/main/java/net/runelite/client/plugins/barbarianassault/BarbarianAssaultPlugin.java]: Yes, it is. The tutorial uses the IN_GAME_BA varbit also. 
After exiting a tutorial there is a 2-3 tick delay for the IN_GAME_BA varbit to change from 1 -> 0 (2-3 ticks of being on the mainland). 
When finishing a real wave the IN_GAME_BA varbit change occurs when it is still in the BA instance. Checking if your in an instance still stops the tutorial from announcing wave duration.

Comment (Adam-) [reply runelite-client/src/main/java/net/runelite/client/plugins/barbarianassault/BarbarianAssaultPlugin.java]: This is the type of things comments were invented for.

**Comment by Adam-** [comment]
@Jacoblairm can you verify this is correct now?

**Comment by Jacoblairm** [comment]
Yes that is correct.

**Pull request merged**